introspect column comments

maktouch opened the pull request:
closes #841

koskimas commented:
How about SQLite and mssql? If they support comments, this feature needs to be implemented for those also.

Tests are missing.

koskimas commented on test/node/src/introspect.test.ts:
```diff
@@ -94,6 +94,7 @@ for (const dialect of DIALECTS) {
                   isNullable: false,
                   isAutoIncrementing: true,
                   hasDefaultValue: true,
+                  column: null
```
Column?

koskimas commented on src/dialect/database-introspector.ts:
```diff
@@ -68,4 +68,5 @@ export interface ColumnMetadata {
   readonly isAutoIncrementing: boolean
   readonly isNullable: boolean
   readonly hasDefaultValue: boolean
+  readonly comment?: string | null
```
Optional is enough. No need to make this also nullable. What would the difference between `undefined` and `null` value be?

koskimas commented on src/dialect/postgres/postgres-introspector.ts:
```diff
@@ -57,6 +57,7 @@ export class PostgresIntrospector implements DatabaseIntrospector {
         'ns.nspname as schema',
         'typ.typname as type',
         'dtns.nspname as type_schema',
+        sql`pg_catalog.col_description(format('%s.%s',isc.table_schema,isc.table_name)::regclass::oid,isc.ordinal_position)`.as('column_description'),
```
I don't think this works. At all. You're referencing tables that don't exist in this query.

maktouch commented:
> Thank you!
> 
> How about SQLite and mssql? If they support comments, this feature needs to be implemented for those also.
> 
> Tests are also needed for every feature. Add at least one test that adds a comment and checks that it gets returned by the introspector. Currently the tests and the whole feature are completely broken.

Haha it's why I made this field optional. I wasn't familiar enough with them to implement it 😂

Ok I'll get familiar, implement the feature, make it non optional but billable, and do the tests.

koskimas commented:
> Haha it's why I made this field optional. I wasn't familiar enough with them to implement it 

It should still be optional even if all built-in dialects support it. Not all databases do, and we want 3rd party dialects to be able to implement the API.

maktouch commented:
> > Haha it's why I made this field optional. I wasn't familiar enough with them to implement it
> 
> It should still be optional even if all built-in dialects support it. Not all databases do, and we want 3rd party dialects to be able to implement the API.

ok that makes sense, so I'll make the value be undefined if it's empty

maktouch commented:
I've reverted my changes and started from scratch in TDD mode
- The first thing I did was modify `test/node/src/test-setup.ts` to add a comment somewhere for each dialect. Unfortunately, I wasn't able to reliably add a comment for sqlite here, so sqlite always returns undefined for comment. If you can help me out here, that would be great! (or we can also just ignore it, it doesn't seem like comment is a real feature in sqlite)
- Then I added the expected comments in `test/node/src/schema.test.ts` and `test/node/src/introspect.test.ts`
- Implemented the features in `src/dialect/$DB/$DB-introspector.ts` until it passed. 

Let me know what else I can change! Thanks! :)

maktouch commented:
Hey @koskimas

Please check this PR when you have time 🙏🙏

Thank you

koskimas reviewed (approved):
Perfect! Thank you 👍
